test/wallet-rpc-test.js

```diff
@@ -0,0 +1,287 @@
+
+    // Assert that the expected test phrase was read
+    // from disk
+    assert.equal(phrase, [
+      'abandon', 'abandon', 'abandon', 'abandon',
+      'abandon', 'abandon', 'abandon', 'abandon',
+      'abandon', 'abandon', 'abandon', 'about'
+    ].join(' '));
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+  });
+
+  describe('getaddressinfo', () => {
+    const watchOnlyWalletId = 'foo';
+    const standardWalletId = 'bar';
+
+    // m/44'/1'/0'/0/{0,1}
+    const pubkeys = [
+      Buffer.from('02a7451395735369f2ecdfc829c0f'
+        + '774e88ef1303dfe5b2f04dbaab30a535dfdd6', 'hex'),
+      Buffer.from('03589ae7c835ce76e23cf8feb32f1a'
+        + 'df4a7f2ba0ed2ad70801802b0bcd70e99c1c', 'hex')
+    ];
+
+    // set up the initial testing state
+    before(async () => {
+      {
+        // Set up the testing environment
+        // by creating a wallet and a watch
+        // only wallet
+        const info = await nclient.getInfo();
+        assert.equal(info.chain.height, 0);
+      }
+
+      {
+        // Create a watch only wallet using the path
+        // m/44'/1'/0' and assert that the wallet
+        // was properly created
+        const accountKey = xpub.xpubkey(network.type);
+        const response = await wclient.createWallet(watchOnlyWalletId, {
+          watchOnly: true,
+          accountKey: accountKey
+        });
+
+        assert.equal(response.id, watchOnlyWalletId);
+
+        const wallet = wclient.wallet(watchOnlyWalletId);
+        const info = await wallet.getAccount('default');
+        assert.equal(info.accountKey, accountKey);
+        assert.equal(info.watchOnly, true);
+      }
+
+      {
+        // Create a wallet that manages the private keys itself
+        const response = await wclient.createWallet(standardWalletId);
+        assert.equal(response.id, standardWalletId);
+
+        const info = await wclient.getAccount(standardWalletId, 'default');
+        assert.equal(info.watchOnly, false);
+      };
+    });
+
+    // The rpc interface requires the wallet to be selected first
+    // the "standardWalletId" is not watch only, and the
+    // "watchOnlyWalletId" is watch only. Select both and assert that
+    // the watch only property that is returned properly matches
+    it('should return iswatchonly correctly', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, false);
+      }
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, true);
+      }
+    });
+
+    // Assert that getaddressinfo returns the same address.
```

Comment:
nit: Pretty much all comments throughout the codebase end with a period, and are generally a sentence or more.

Reply:
There are more of these, not just this case.